File: app/services/notification_service.py
from app.database import SessionLocal
from app.models.monitoring import SystemNotification
import sqlalchemy as sa

class NotificationService:
    @staticmethod
    def notify(title, message, level="info", source="system"):
        db = SessionLocal()
        notif = SystemNotification(
            title=title,
            message=message,
            level=level,
            source=source
        )
        db.add(notif)
        db.commit()
        db.close()
        # In the future, we could trigger WebSockets/Emails here

    @staticmethod
    def get_unread_count():
        db = SessionLocal()
        count = db.scalar(sa.select(sa.func.count(SystemNotification.id)).where(SystemNotification.is_read == False))
        db.close()
        return count or 0

notification_service = NotificationService()

File: crm-portal-ts/src/services/notification_service.ts
import { SessionType } from "typeorm/session";
import { SystemNotificationInterface as NotificationModel} from 'app.models'; // Assuming you have a model named `SystemNotification` in your app's database module 

class DatabaseService{  
    private session: SessionType;    
      
    async initialize(){        
        this.session = await createConnection().then(connection => connection);         
    }     
}          
            
export class NotificationService {           
    dbSvc : DatabaseService ;  // Assuming you have a database service instance named `db` in your application  
    
    constructor(){        this.dbSvc = new DatabaseService();         await this.dbSvc.initialize() }     
      
    async notify(title: string, message:string , level="info", source='system'){         
           let notif : NotificationModel  = {  // Assuming you have a model named `SystemNotification` in your app  
            title: title,            
            message:message,             
            level:level,              
            source:source               
        }    ;         this.session.getRepository(NotificationModel).save(notif);     await  (this as any) .dbSvc;          // Assuming you have a database service instance named `db` in your application  
           return notif;}      async function get_unread_count(){       let count =await this.session            
            .getRepository(NotificationModel).find({where:{isRead : false}});     await (this as any)  . dbSvc;          // Assuming you have a database service instance named `db` in your application   return    !count?0:(typeof count ==='number') ? count.length || 0 ; }     
}           const notification_service = new NotificationService();         (async function(){ await  notif!=null;}